File: src/main/java/me/tbg/match/bot/BotConfig.java
package me.tbg.match.bot;

import org.bukkit.configuration.Configuration;

public class BotConfig {

    private boolean enabled;

    private String token;
    private String serverId;
    private String matchChannel;
    private String serverName;
    private String mapImageNotFound;

    public BotConfig(Configuration config) {
        reload(config);
    }

    public void reload(Configuration config) {
        this.enabled = config.getBoolean("enabled");
        this.token = config.getString("token");
        this.serverId = config.getString("server");
        this.matchChannel = config.getString("match-channel");
        this.serverName = config.getString("server-name");
        this.mapImageNotFound = config.getString("map-image-not-found");
    }

    public boolean isEnabled() {
        return enabled;
    }

    public String getToken() {
        return token;
    }

    public String getServerId() {
        return serverId;
    }

    public String getMatchChannel() {
        return matchChannel;
    }

    public String getServerName() { return serverName; }

    public String getMapImageNotFound() {
        return mapImageNotFound;
    }
}

File: src/main/java/me/tbg/match/bot/BotConfig.kt
package me.tbg.match.bot

import org.bukkit.configuration.Configuration

class BotConfig(config: Configuration) {
    var isEnabled = false
        private set
    var token: String? = null
        private set
    var serverId: String? = null
        private set
    var matchChannel: String? = null
        private set
    var mapImagesURL: String? = null
        private set

    fun reload(config: Configuration) {
        isEnabled = config.getBoolean("enabled")
        token = config.getString("token")
        serverId = config.getString("server")
        matchChannel = config.getString("match-channel")
        mapImagesURL = config.getString("map-images-url")
    }

    init {
        reload(config)
    }
}